public/app/features/trails/MetricSelect/MetricSearchTermsVariable.ts

```diff
@@ -0,0 +1,52 @@
+import { VariableHide } from '@grafana/data';
+import { VariableValue, SceneVariableState, sceneGraph, AdHocFiltersVariable, TextBoxVariable } from '@grafana/scenes';
+
+import { VAR_FILTERS, VAR_METRIC_SEARCH_TERMS } from '../shared';
+
+import { createPromRegExp, deriveSearchTermsFromInput } from './util';
+
+export interface MetricsSearchTermsVariableState extends SceneVariableState {
+  terms: string[];
+}
+
+export class MetricSearchTermsVariable extends TextBoxVariable {
+  constructor(initialState: Partial<{ value: string }>) {
+    super({
+      value: '',
+      hide: VariableHide.hideVariable,
+      name: VAR_METRIC_SEARCH_TERMS,
+      ...initialState,
+    });
+  }
+
+  getValue(): VariableValue {
+    const filtersVar = sceneGraph.lookupVariable(VAR_FILTERS, this);
+
+    const { value } = this.state;
+    let separator = ',';
+
+    if (filtersVar instanceof AdHocFiltersVariable) {
+      const value = filtersVar.getValue()?.valueOf();
+      if (!value) {
+        separator = '';
+      }
+    }
+
+    return {
+      formatter: (format: Format) => {
+        switch (format) {
+          case 'filter': {
+            const terms = deriveSearchTermsFromInput(value);
+            const regex = createPromRegExp(terms);
+            const result = !terms?.length ? '' : `${separator}__name__=~"${regex}"`;
+            return result;
+          }
+          default:
+            return value;
+        }
+      },
+    };
+  }
+}
```

Comment:
just not sure I understand what is going on here. 

I think it would be easier to just represent / store it as a search string, but parse it into terms and build the regex on the fly (in a function called from _refreshMetricNames), think representing this as a custom variable with custom formatter could just be complicating things.

Reply:
Seems to be the case. It was an interesting exploration into making variables, but on the most part things are much cleaner with a `metricSearch` state property on DataTrail.


Here is the new implementation of the content you highlighted here;

https://github.com/grafana/grafana/pull/87884/files#diff-7b6fff7b7211c87207d063b18a76f1fb5bc7e9aadb4b05e3cf86d0debbadf8eeR163-R184